lib/handlers/router-entities/pool-caching/v3/dynamo-pool-provider.ts

```diff
@@ -0,0 +1,109 @@
+import { ChainId, IV3PoolProvider, V3PoolAccessor } from '@uniswap/smart-order-router'
+import { Token } from '@uniswap/sdk-core'
+import { ProviderConfig } from '@uniswap/smart-order-router/build/main/providers/provider'
+import { FeeAmount, Pool } from '@uniswap/v3-sdk'
+import { DynamoDB } from 'aws-sdk'
+
+export class DynamoPoolProvider implements IV3PoolProvider {
+  private readonly ddbClient: DynamoDB.DocumentClient
+  private readonly tableName: string
+  private readonly ttlMinutes: number
+  private POOL_CACHE_KEY = (chainId: ChainId, address: string) => `pool-${chainId}-${address}`
+
+  constructor(protected chainId: ChainId, protected poolProvider: IV3PoolProvider) {
```

Comment:
you might need to receive the table name here, roght?

Reply:
Yes from [cache-dynamo-pool.ts](https://github.com/Uniswap/routing-api/pull/201/files#diff-5339f16f1f7af49f5b71baf15da11c9f3f02fb225346eea97ececdf1c3fb42b2)